File: Assets/PointHandler.cs
using UnityEngine;
using System.Collections;

public class PointHandler : MonoBehaviour {

    private bool addPoint = false;
    private bool addLink  = false;
    private bool isLocked;
    public GameObject points;

    void Start(){

    }

    private Vector3 mouseWorldSpace;

    void Update(){
        mouseWorldSpace = Camera.main.ScreenToWorldPoint(Input.mousePosition);

        if (Input.GetMouseButtonDown(0) && addPoint == true)
        {
            addPoint = false;
            Instantiate(points, new Vector3(mouseWorldSpace.x, mouseWorldSpace.y, 0f), transform.rotation);
        }
        else if (Input.GetMouseButtonDown(0) && addLink == true)
        {
            addLink = false;
            Instantiate(points, new Vector3(mouseWorldSpace.x, mouseWorldSpace.y, 0f), transform.rotation);
        }
    }

    void OnAddClick()
    {
        addPoint = true;
    }

    GameObject[] len;

    void OnLinkClick(GameObject target){
        var link = Instantiate(points, new Vector3(mouseWorldSpace.x, mouseWorldSpace.y, 0f), transform.rotation) as GameObject;
        len = GameObject.FindGameObjectsWithTag("Point");
        bool pointWithNameFound = false;
        for (int x = 0; x < len.Length; x++){
            for (int z = 0; z < len.Length; z++){
                if (GameObject.FindGameObjectsWithTag("Point")[z].name == ("Point" + (x + 1))){
                    pointWithNameFound = true;
                }
            }
            if (!pointWithNameFound)
            {
                link.name = "Point" + (x + 1);
            }
            pointWithNameFound = false;
        }
        DrawLine liner = link.GetComponent<DrawLine>();
        liner.target = target;
    }
}

File: Assets/Not used/PointHandler.js
#pragma strict

private var addPoint : boolean = false;
private var addLink : boolean = false;
public var points : Transform;

function Start () {

}

function Update () {
	if(Input.GetMouseButtonDown(0) && addPoint == true){
		Instantiate(points, new Vector3(Input.mousePosition.x, Input.mousePosition.y, 0f), transform.rotation);
	}
	else if(Input.GetMouseButtonDown(0) && addLink == true){
		Instantiate(points, new Vector3(Input.mousePosition.x, Input.mousePosition.y, 0f), transform.rotation);
	}
}

function OnAddClick(){
	yield WaitForSeconds(0.02f);
	addPoint = true;
}

function OnLinkClick(){
	yield WaitForSeconds(0.02f);
	addLink = true;
}